chall02/jcervill.py

```diff
@@ -0,0 +1,82 @@
+import sys
+
+i = 0
+morse = {
+	'a':'.-',
+	'A':'.-',
+	'b':'-...',
+	'B':'-...',
+	'c':'-.-.',
+	'C':'-.-.',
+	'd':'-..',
+	'D':'-..',
+	'e':'.',
+	'E':'.',
+	'f':'..-.',
+	'F':'..-.',
+	'g':'--.',
+	'G':'--.',
+	'h':'....',
+	'H':'....',
+	'i':'..',
+	'I':'..',
+	'j':'.---',
+	'J':'.---',
+	'k':'-.-',
+	'K':'-.-',
+	'l':'.-..',
+	'L':'.-..',
+	'm':'--',
+	'M':'--',
+	'n':'-.',
+	'N':'-.',
+	'o':'---',
+	'O':'---',
+	'p':'.--.',
+	'P':'.--.',
+	'q':'--.-',
+	'Q':'--.-',
+	'r':'.-.',
+	'R':'.-.',
+	's':'...',
+	'S':'...',
+	't':'-',
+	'T':'-',
+	'u':'..-',
+	'U':'..-',
+	'v':'...-',
+	'V':'...-',
+	'w':'.--',
+	'W':'.--',
+	'x':'-..-',
+	'X':'-..-',
+	'y':'-.--',
+	'Y':'-.--',
+	'z':'--..',
+	'Z':'--..',
+	' ':' '
+}
+
+def error():
+	print("usage: " + sys.argv[0] + " <a-zA-Z string>")
+	exit(1)
+
+def is_morse(dictionary, strin):
+	for j in strin:
+		if not (j.isspace() or j.isalpha()):
+			error()
+		if not(dictionary[j]):
+			error()
```

Comment:
This part of the code is never reached, as your previous condition is englobes the content of the dictionary.
```suggestion
```
Also, this is not the correct way of doing this check, for various reason:
- `dictionnary[j]` will raise a `KeyError` exception when the value is absent
- `not dictionnary[j]` will be true if `dictionnary[j]` equals `""`, `0`, `{}`, `[]` and many other values

The correct form would have been:
```suggestion
		if dictionnary.get(j) is None:
			error()
```
Assuming `None` is never to be part of the dictionary's values.